### ISSUE TITLE
[prim/rtl] Does prim_sync_reqack support different reset domain for src and dst?

### ISSUE THREAD

**Issue by cindychip** [lowRISC/opentitan#14086]
Hi Michael, Pirmin, and Tim,

When I implemented lc_ctrl transition with kmac reset test, I saw that prim_sync_reqack sent an extra kmac data request after kmac handshake is done. Though LC transition finishes without any error, but I still want to double check the behavior with you.

Sequence:
1). Send a LC_CTRL transition request
2). Wait until LC_CTRL sends a kmac data request to scramble tokens
3). Issue sys domain reset to reset kmac but does not reset LC_CTRL
4). LC_CTRL resends kmac data request
5). Wait until KMAC finishes scrambling and returns scrambled tokens
6). Issue another sys domain reset
7). LC finishes state transition without any error. But in the waves, I saw LC_CTRL sends out another kmac data request.

I attached the waves below, but I think the issue comes from the [prim_sync_reqack](https://github.com/lowRISC/opentitan/blob/master/hw/ip/prim/rtl/prim_sync_reqack.sv) module.
For this case, LC_CTRL is the src and KMAC is the dst domain.
When the first KMAC and LC_CTRL handshake is done at step 5), both `src_fsm_cs` and `dst_fsm_cs` are at `ODD` state. However, when we issue another `dst_rst_ni`, the `dst_fsm_cs` is reverted back to `EVEN` state again. I think that somehow messed up with the internal `dst_req` polarity and ended up sending an extra `dst_req`.

Again, there is no impact for LC state transition as far as I can tell from the results. But just want to raise this up in case you have any concerns. Please ping me if you need a smart-id to look at the waves.

![image](https://user-images.githubusercontent.com/11466553/183581047-042bfa9e-2c36-4abb-8751-d16b4c0a022b.png)

Thanks,
Cindy

**Comment by tjaychen** [comment]
yeah this is an issue, it's popped up a few other areas as well. 
the cdc modules in prim can't "really" (related #14072) be reset one side at a time. The reset really needs to be the "same" root, but different clock domain. 

can you remind me where the `prim_sync_reqack` is located in the hierarchy? I want to take a closer look at a couple of things.

**Comment by cindychip** [comment]
Sure! Thanks for the feedback -  it is located here:
https://github.com/lowRISC/opentitan/blob/master/hw/ip/lc_ctrl/rtl/lc_ctrl_kmac_if.sv#L76
https://github.com/lowRISC/opentitan/blob/master/hw/ip/prim/rtl/prim_sync_reqack_data.sv#L45

**Comment by tjaychen** [comment]
sorry i'm jumping around.
I responded in the other thread, I agree with you guys this is probably acceptable behavior. 
But it would be nice to fix the behavior from prim_sync_reqack where it creates ghost transactions (even if the macro behavior is acceptable). 

This mostly just means we need to wire up a different reset on the interface, i can have a look at this.

**Comment by tjaychen** [comment]
will be fixed by #15317
